charts/tfy-configs/files/cicd-templates/gitlab-pipelines-self-build-image.yaml

```diff
@@ -0,0 +1,80 @@
+name: Build Image Yourself and Deploy Image on TrueFoundry
+cicd_provider_id: gitlab
+enabled: true
+description: "Build Docker Image using your own steps and deploy the image on TrueFoundry."
+deployment_mode: deploy
+build_source: local
+image_builder: self
+steps:
+  - label: Generate API Key
+    icon: null
+    usage: Generate an API Key to authenticate and deploy applications
+    type: generate-api-key
+  - label: Add API Key to Secrets
+    icon: null
+    usage: null
+    type: markdown-content
+    args:
+      content: |
+        In your GitLab Repository, navigate to **Settings > CI/CD** then expand **Variables**.
+        Add a new secret called `TFY_API_KEY` and set the generated api key as value and select the **Mask Variable** checkbox.
+  - label: Download Application Spec
+    icon: null
+    usage: Click the button below to download the `truefoundry.yaml` application spec file. Copy it to the root of your project directory.
+    type: download-truefoundry-spec
+  - label: Create GitHub Action
+    icon: null
+    usage: |
+      Add the below workflow as `.gitlab-ci.yml` in your root directory.
+      Following Pipelines will be triggered on each push to `main` branch
+    type: markdown-content
+    args:
+      content: |
+        ```yaml
+        stages:
+          - build
+          - deploy
+        
+        workflow:
+          rules:
+            - if: $CI_COMMIT_BRANCH == "main"
+
+        variables:
+          IMAGE_REPO: $DOCKER_IMAGE_REPO
```

Comment:
```suggestion
          # Please add your `REGISTRY` and `REPO_NAME` here to push the image to
          REGISTRY: <YOUR DOCKER REGISTRY HERE>
          REPO_NAME: $CI_PROJECT_NAME
          IMAGE_REPO: $REGISTRY/$REPO_NAME
          IMAGE_TAG: $CI_COMMIT_SHA
          IMAGE_URI: $IMAGE_REPO:$IMAGE_TAG
```

Reply:
Also please check if this works, like can I declare a variable and use it immediately in the next line?